sw/device/lib/testing/flash_ctrl_testutils.c

```diff
@@ -311,23 +311,26 @@ status_t flash_ctrl_testutils_backdoor_init(
                                                     /*he_en*/ false);
 }
 
-status_t flash_ctrl_testutils_backdoor_wait_update(
-    dif_flash_ctrl_state_t *flash_state, uintptr_t addr, size_t timeout) {
-  static uint32_t data = UINT32_MAX;
-  TRY(flash_ctrl_testutils_write(
-      flash_state, (uint32_t)addr - TOP_EARLGREY_FLASH_CTRL_MEM_BASE_ADDR, 0,
-      &data, kDifFlashCtrlPartitionTypeData, 1));
-  IBEX_TRY_SPIN_FOR(UINT32_MAX != *(uint32_t *)addr, timeout);
-  return OK_STATUS();
-}
-
-status_t flash_ctrl_testutils_flush_read_buffers(void) {
-  static volatile const uint32_t kFlashFlusher[8];
-  // Cause read buffers to flush.
-  uint32_t count = 0;
+static void flash_ctrl_testutils_flush_read_buffers(void) {
+  // Cause read buffers to flush since it reads 32 bytes, which is the
+  // size of the read buffers.
+  enum { kBufferBytes = 32 };
+  static volatile const uint8_t kFlashFlusher[kBufferBytes];
   for (int i = 0; i < sizeof(kFlashFlusher); ++i) {
-    count += kFlashFlusher[i];
+    kFlashFlusher[i];
```

Comment:
Is this sufficient to read flash?

Reply:
The static array kFlashFlusher is mapped into flash, and it is volatile, so the compiler should honor reads. I added a void cast, but all tests are passing and the read finds changed data promptly after the backdoor overwrite, so this is working.

Reply:
Good to know.